packages/algob/src/lib/msig.ts

```diff
@@ -89,3 +89,31 @@ export function validateMsig (msig: MultiSig | undefined): void {
     throw new Error("Error fetching multisigned logic signature from file - invalid/undefined msig");
   }
 }
+
+/**
+ * Description: Signs an encoded signed multi-sig transaction object
+ * @param {Account} signerAccount
+ * @param {TxSig} signedTxn
+ * @returns {TxSig} signed transaction object
+ */
+export function signMultiSig (signerAccount: Account, signedTxn: TxSig): TxSig {
+  const tx = signedTxn;
+  const decodedTx = decodeSignedTransaction(tx.blob);
+  console.debug("Decoded txn: %O", decodedTx);
+  validateMsig(decodedTx.msig);
+  console.log("Msig: %O", decodedTx.msig);
+  const addresses = [];
+  for (var sig of decodedTx.msig.subsig) {
+    addresses.push(encodeAddress(Uint8Array.from(sig.pk)));
+  }
+  const mparams = {
+    version: decodedTx.msig.v,
+    threshold: decodedTx.msig.thr,
+    addrs: addresses
+  };
+  signedTxn = appendSignMultisigTransaction(tx.blob, mparams, signerAccount.sk);
+  const decodedSignedTxn = decodeSignedTransaction(signedTxn.blob);
```

Comment:
Why are we decoding tx after signing? Won't user need the signed tx (msgpack encoded) only to use furthur?

Reply:
We are decoding it to show the final msig and transaction object.